timemachine/fe/mcgregor.py

```diff
@@ -128,6 +128,30 @@ def __init__(self, n_vertices, edges):
             for edge_idx in edges:
                 self.ve_matrix[vertex_idx][edge_idx] = True
 
+    def _verify_partially_connected(self, source, node_states, n_mapped_nodes):
+        """A fast BFS node generator, adapted from _plain_bfs in networkx."""
+        unseen = [1] * self.n_vertices
+        unseen[source] = 0
+        nextlevel = [source]
+        mapped_count = 1
+
+        if mapped_count == n_mapped_nodes:
+            return True
+
+        while nextlevel:
+            thislevel = nextlevel
+            nextlevel = []
+            for v in thislevel:
+                for w in self.get_neighbors(v):
+                    if node_states[w] and unseen[w]:
```

Comment:
nit: would prefer
```suggestion
                    if node_states[w] != NODE_STATE_DEMAPPED and unseen[w]:
```
to avoid coupling the logic with the representation of the enum

Reply:
This is an extremely hot-loop, your suggestion was actually my old implementation (which requires an additional comparison operator) - similar to why we do `unseen[w]` vs just `not seen[w]` (as in the `_plain_bfs()` implementation)

Reply:
Hmm, it seems pretty surprising that the existing version should be significantly faster... In my benchmarks, the more explicit version is slightly faster

Reply:
you're right - I just tested this again and it seems to be the same. I'll make the comparison explicit.